src/SonarScanner.MSBuild.TFS/TrxFileReader.cs

```diff
@@ -82,153 +82,132 @@ public TrxFileReader(ILogger logger)
+            else
             {
-                return null;
+                this.logger.LogInfo(Resources.TRX_DIAG_TrxFilesFound, string.Join(", ", trxFiles));
             }
 
-            switch (attachmentUris.Count())
-            {
-                case 0:
-                    logger.LogDebug(Resources.TRX_DIAG_NoCodeCoverageInfo);
-                    return null;
-
-                case 1:
-                    var attachmentName = attachmentUris.First();
-                    logger.LogDebug(Resources.TRX_DIAG_SingleCodeCoverageAttachmentFound, attachmentName);
-
-                    var trxDirectoryName = Path.GetDirectoryName(trxFilePath);
-                    var trxFileName = Path.GetFileNameWithoutExtension(trxFilePath);
-
-                    var possibleCoveragePath =
-                        new[]
-                        {
-                            attachmentName,
-                            Path.Combine(trxDirectoryName, trxFileName, "In", attachmentName),
-                            // https://jira.sonarsource.com/browse/SONARMSBRU-361
-                            // With VSTest task the coverage file name uses underscore instead of spaces.
-                            Path.Combine(trxDirectoryName, trxFileName.Replace(' ', '_'), "In", attachmentName)
-                        }
-                        .FirstOrDefault(path => File.Exists(path));
-
-                    if (possibleCoveragePath != null)
-                    {
-                        logger.LogDebug(Resources.TRX_DIAG_AbsoluteTrxPath, possibleCoveragePath);
-                    }
-                    else
-                    {
-                        logger.LogWarning(Resources.TRX_WARN_InvalidConstructedCoveragePath, trxFilePath, attachmentName);
-                    }
-
-                    return possibleCoveragePath;
-
-                default:
-                    logger.LogWarning(Resources.TRX_WARN_MultipleCodeCoverageAttachmentsFound, string.Join(", ", attachmentUris.ToArray()));
-                    return null;
-            }
+            return trxFiles;
         }
 
-        private bool TryExtractCoverageFilePaths(string trxFilePath, out IEnumerable<string> coverageFilePaths)
+        private bool TryExtractCoverageFilePaths(IEnumerable<string> trxFilePaths, out IEnumerable<string> coverageFilePaths)
         {
-            Debug.Assert(File.Exists(trxFilePath));
-
-            coverageFilePaths = null;
-            var continueProcessing = true;
-
             var runAttachments = new List<string>();
-            try
+
+            foreach (var trxPath in trxFilePaths)
             {
-                var doc = new XmlDocument();
-                doc.Load(trxFilePath);
-                var nsmgr = new XmlNamespaceManager(doc.NameTable);
-                nsmgr.AddNamespace("x", CodeCoverageXmlNamespace);
+                try
+                {
+                    var doc = new XmlDocument();
+                    doc.Load(trxPath);
```

Comment:
Bearing in mind what Lars said about XXE vulnerabilities this morning, I had a quick look at the [XXE cheat sheet](https://www.owasp.org/index.php/XML_External_Entity_(XXE)_Prevention_Cheat_Sheet#XmlDocument) to check whether this code is vulnerable. I think we're ok because we're targeting .NET 4.6+ and NET Standard 2.0+.
However, it wouldn't hurt to explicitly set _doc.XmlResolver = null_ and include a comment saying this code has been checked and is safe from  XXE (possibly including a link to the OWASP cheat sheet).

Reply:
I don't think this should be handled in this PR.